src/BlazorWebView/src/core/iOS/BlazorWebViewHandler.iOS.cs

```diff
@@ -0,0 +1,267 @@
+			}
+
+			public void DidReceiveScriptMessage(WKUserContentController userContentController, WKScriptMessage message)
+			{
+				if (message is null)
+				{
+					throw new ArgumentNullException(nameof(message));
+				}
+				_messageReceivedAction(new Uri(AppOrigin), ((NSString)message.Body).ToString());
+			}
+		}
+
+		private class SchemeHandler : NSObject, IWKUrlSchemeHandler
+		{
+			private readonly BlazorWebViewHandler _webViewHandler;
+
+			public SchemeHandler(BlazorWebViewHandler webViewHandler)
+			{
+				_webViewHandler = webViewHandler;
+			}
+
+			[Export("webView:startURLSchemeTask:")]
+			public void StartUrlSchemeTask(WKWebView webView, IWKUrlSchemeTask urlSchemeTask)
+			{
+				var responseBytes = GetResponseBytes(urlSchemeTask.Request.Url.AbsoluteString, out var contentType, statusCode: out var statusCode);
+				if (statusCode == 200)
+				{
+					using (var dic = new NSMutableDictionary<NSString, NSString>())
+					{
+						dic.Add((NSString)"Content-Length", (NSString)(responseBytes.Length.ToString(CultureInfo.InvariantCulture)));
+						dic.Add((NSString)"Content-Type", (NSString)contentType);
+						// Disable local caching. This will prevent user scripts from executing correctly.
+						dic.Add((NSString)"Cache-Control", (NSString)"no-cache, max-age=0, must-revalidate, no-store");
+						using var response = new NSHttpUrlResponse(urlSchemeTask.Request.Url, statusCode, "HTTP/1.1", dic);
+						urlSchemeTask.DidReceiveResponse(response);
+					}
+					urlSchemeTask.DidReceiveData(NSData.FromArray(responseBytes));
+					urlSchemeTask.DidFinish();
+				}
+			}
+
+			private byte[] GetResponseBytes(string url, out string contentType, out int statusCode)
+			{
+				var allowFallbackOnHostPage = url.EndsWith("/");
+				if (_webViewHandler._webviewManager!.TryGetResponseContentInternal(url, allowFallbackOnHostPage, out statusCode, out var statusMessage, out var content, out var headers))
+				{
+					statusCode = 200;
+					using var ms = new MemoryStream();
+					var uri = new Uri(url);
+
+					content.CopyTo(ms);
+
+					content.Position = 0;
+					var sr = new StreamReader(content);
+					var cc = sr.ReadToEnd();
+
+					content.Dispose();
+
+
+					var headersDict =
+						new Dictionary<string, string>(
+						headers
+							.Split(Environment.NewLine)
+							.Select(headerString =>
+								new KeyValuePair<string, string>(
+									headerString.Substring(0, headerString.IndexOf(':')),
+									headerString.Substring(headerString.IndexOf(':') + 2))));
+
+					contentType = headersDict["Content-Type"];
+					if (cc.StartsWith("<!DOCTYPE", StringComparison.Ordinal))
+					{
+						contentType = "text/html";
+					}
+					else if (cc.Contains("box-shadow", StringComparison.Ordinal))
+					{
+						contentType = "text/css";
+					}
+					else if (url.Contains(".js", StringComparison.Ordinal))
+					{
+						contentType = "application/javascript";
+					}
```

Comment:
Are we working around some thing here?

Reply:
This feels fragile

Reply:
Yes this is all temporary until we pick up the latest Blazor assemblies.

Reply:
Everything to do with MIME types and headers goes away.